plugins/groovy/groovy-psi/src/org/jetbrains/plugins/groovy/geb/GebUtil.java

```diff
@@ -37,12 +45,81 @@ public static boolean contributeMembersInsideTest(PsiScopeProcessor processor,
 
       if (pageClass != null) {
         if (!pageClass.processDeclarations(processor, state, null, place)) return false;
+        contributePageContent(processor, state, place);
       }
     }
 
     return true;
   }
 
+  private static void contributePageContent(PsiScopeProcessor processor, ResolveState state, PsiElement place) {
+    if(place instanceof GrReferenceExpressionImpl expr
+       && !(expr.getParent() instanceof GrMethodCall)) {
+
+      PsiClassType gebPageType = PsiType.getTypeByName("geb.Page", place.getProject(), place.getResolveScope());
+      PsiClass ourPage = findPageChange(expr, gebPageType);
+      if (ourPage != null) {
+        Map<String, PsiClass> supers = ClassUtil.getSuperClassesWithCache(ourPage);
+        String nameHint = ResolveUtil.getNameHint(processor);
+
+        for (PsiClass psiClass : supers.values()) {
+          Map<String, PsiMember> contentElements = getContentElements(psiClass);
+
+          if (nameHint == null) {
+            for (Map.Entry<String, PsiMember> entry : contentElements.entrySet()) {
+              processor.execute(entry.getValue(), state);
+            }
+            return;
+          }
+          else {
+            PsiVariable defElement = (PsiVariable)contentElements.get(nameHint);
+            if (defElement != null) {
+              processor.execute(defElement, state);
+            }
+          }
+        }
+      }
+    }
+  }
+
+  
+  private static @Nullable PsiClass findPageChange(PsiElement place, PsiClassType pageType) {
+    PsiElement currentElement = place;
+    PsiClass ourPage = null;
+    while (currentElement != null) {
```

Comment:
It seems that this method will be invoked for each reference inside the method, but you actually traverse the whole method starting from the innermost block to the outermost. Could you, please, rewrite it in the way like:
```
GrCodeBlock block = PsiTreeUtil.getParentOfType(place, GrCodeBlock.class):
for (var statement : block.getStatements()) {
   ...
}
```